static/js/components/UserInfoCard.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import React from 'react';
+import Grid, { Cell } from 'react-mdl/lib/Grid';
+import { Card } from 'react-mdl/lib/Card';
+import IconButton from 'react-mdl/lib/IconButton';
+
+import ProfileImage from './ProfileImage';
+import {
+  getEmployer,
+  getPreferredName,
+  userPrivilegeCheck
+} from '../util/util';
+import type { Profile } from '../flow/profileTypes';
+
+export default class UserInfoCard extends React.Component {
+  props: {
+    profile: Profile,
+    toggleShowPersonalDialog: () => void
+  };
+
+  getCurrentWorkPlace: Function = (profile: Profile): string => {
+    let companyName = "";
+    if (!getEmployer(profile).isNothing) {
+      companyName = getEmployer(profile).value;
+    }
+    return companyName;
+  }
```

Comment:
There's a utility function you should use here instead, `getEmployer`, from `util.js`.